### PR TITLE
[ownership] Change LLChecker APIs that use an out array to return leaked blocks to caller to instead use a callback.

### PR DESCRIPTION
**Pull request by gottesmm**
NFC.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

File: include/swift/SIL/LinearLifetimeChecker.h [hunk 1]
```diff
@@ -107,10 +95,19 @@ class LinearLifetimeChecker {
   /// error.
   /// \p leakingBlocks If non-null a list of blocks where the value was detected
   /// to leak. Can be used to insert missing destroys.
```
Comment (atrick) [inline include/swift/SIL/LinearLifetimeChecker.h]: stale comments

File: include/swift/SIL/LinearLifetimeChecker.h [hunk 2]
```diff
@@ -74,22 +74,10 @@ class LinearLifetimeChecker {
   /// Returns true if we completed the consuming use set and discovered that \p
   /// consumingUse is not strongly control equivalent to value (meaning
   /// consumingUse is not in the same loop in the loop nest as value).
```
Comment (atrick) [inline include/swift/SIL/LinearLifetimeChecker.h]: It's not part of this PR, but I have to say: convention is to return `true` when the analysis succeeds. If the consumingUse cannot be used to end the lifetime, then that would be a failure of the analysis. It's nice to show an example of why that may be (use inside a loop), but I don't understand why "control equivalence" of the use would be necessary.

**Review by atrick — APPROVED** [review]
Great! I have comments on the comments.

**Comment by gottesmm** [comment]
I'll hit those comments once this lands.

**Pull request merged**